src/main/groovy/com/ofg/twitter/place/extractor/PlacesExtractor.groovy

```diff
@@ -21,8 +22,12 @@ class PlacesExtractor {
 
     Map<String, Optional<Place>> extractPlacesFrom(List<Tweet> tweets) {
         Map<String, Optional<Place>> foundPlaces = new ConcurrentHashMap<>()
-        GParsPool.withPool {
-            tweets.eachParallel { foundPlaces << appendExtractedTweet(it) }
+        String savedCorrelationId = CorrelationIdHolder.get()
+        GParsPool.withPool(1) {
+            tweets.eachParallel {
```

Comment:
A completely transparent would be great, but I don't know [1] how we could globally hook into a place when a code fragment is assigned to a thread for execution to manage the correlation id.

[1] - it could be possible with AOP, but the end application would have to be run with AspectJ agent to instrumentalize JDK code.

Btw, Tomek, you commented outdated code. In current version it is required only to wrap a closure with another closure using a utility method (without a direct reference to `CorrelationHolder`):
```
tweets.eachParallel closureWithId {
    foundPlaces << appendExtractedTweet(it)
}
```